wikibase/queryService/api/Sparql.js

```diff
@@ -238,6 +234,28 @@ wikibase.queryService.api.Sparql = ( function ( $ ) {
 			}
 		} catch ( e ) {
 		}
+		
+		//if the extraction from the server response fails
+		if(error.code === ERROR_CODES.UNKNOWN) {			
+			if ( request.status === 0 || exception ) {
+				error.code = ERROR_CODES.SERVER;
+				error.message = exception.message;
+			} else if (request.status === 400) {
+				error.code = ERROR_CODES.MALFORMED;
+			} else if ((request.status === 500 || request.status === 504) && 
+				(error.debug.toLowerCase().includes("timeout") || error.debug.toLowerCase().includes("timed out") || 
+				error.debug.toLowerCase().includes("time out") || error.debug.toLowerCase().includes("time-out"))) {
+				error.code = ERROR_CODES.TIMEOUT;
+				error.debug = error.debug + "\n\n" +
+				"The query took too long to run, likely due to resource-intensive operations such as DISTINCT or COUNT operations over a large \n" +
+				"result set, or you used complex filter operations like regex. Try running the query again, and if the issue persists, consider \n" +
```

Comment:
there should be no hardcoded message here, it should go to i8n messages

Reply:
we would also need messages for all languages, eventually by using google translate or other service, I supose the wikimedia team uses autoated translation for the i8n messages